### PR TITLE
test(cardinal): add test for create, start, stop, purge

### PR DESCRIPTION
**Pull request by zulkhair**
Closes: WORLD-362

## Overview
  
Adding unit test for create, start, stop, purge command

## Brief Changelog
  
- Added new test that cover create, start, stop, and purge command

## Testing and Verifying

Test will do this scenario
- run create command to clone starter-game-template
- start the cardinal
- hit `/health` endpoint and verify the response
- stop the cardinal
- hit `/health` endpoint and make sure it will be error
- purge the cardinal

### CODE REVIEW

**Comment by zulkhair** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/32?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #28** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/28?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #32** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Argus-Labs/world-cli/32?utm_source=stack-comment).

File: cmd/world/root/create.go [hunk 1]
```diff
@@ -22,21 +23,26 @@ const TemplateGitUrl = "https://github.com/Argus-Labs/starter-game-template.git"
...
-			return err
-		}
-		return nil
-	},
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger.SetDebugMode(cmd)
+			p := tea.NewProgram(NewWorldCreateModel(args), tea.WithOutput(writer))
+			if _, err := p.Run(); err != nil {
+				fmt.Println(err)
```
Comment (jerargus) [inline cmd/world/root/create.go:38]: Doesn't cobra print errors to the screen? Is this fmt.Println necessary?

File: cmd/world/root/root_test.go [hunk 2]
```diff
@@ -95,3 +98,54 @@ func TestExecuteDoctorCommand(t *testing.T) {
...
+	assert.Error(t, err,
+		"Get \"http://127.0.0.1:3333/health\": dial tcp 127.0.0.1:3333: connect: connection refused")
+
+	// Purge cardinal
+	rootCmd.SetArgs([]string{"cardinal", "purge"})
+	err = rootCmd.Execute()
+	assert.NilError(t, err)
+
+	// Remove dir
+	os.RemoveAll(gameDir)
```
Comment (jerargus) [inline cmd/world/root/root_test.go:150]: nit: This should be moved to just below MkdirTemp:
`defer os.RemoveAll(gameDir)`

**Review by jerargus — APPROVED** [review]

File: cmd/world/root/create.go (hunk 1 above)
Comment (zulkhair) [reply cmd/world/root/create.go:38]: this error is from bubletea lib not cobra. so yes this is necessary.

File: cmd/world/root/root_test.go (hunk 2 above)
Comment (zulkhair) [reply cmd/world/root/root_test.go]: updated

File: cmd/world/root/create.go (hunk 1 above)
Comment (zulkhair) [reply cmd/world/root/create.go:38]: ahhh, sorry. this is inside cobra. yes it will be printed by cobra.
will remove this line.

**Pull request merged**